test/users/edit_user_info_test.js

```diff
@@ -171,5 +171,67 @@ describe("Users", function () {
                 return expect(updateUser({}, { password: "" })).to.be.an.api.error(400, "password_required");
             });
         });
+
+        describe("changing npi as user", function () {
+            var request;
+            beforeEach(function () {
+                request = updateUser({role: "user" }, { npi: "1245319599" });
+            });
+
+            it("returns a successful response", function () {
+                return expect(request).to.be.a.user.success;
+            });
+            it("does not revoke our access tokens", function () {
+                return request.then(auth.checkTokenWorks(token));
+            });
+            it("stills let us authenticate", function () {
+                return request.then(function () {
+                    return expect(tokenEndpoint(credentials)).to.be.an.api.postSuccess;
+                });
+            });
+
+            it("doesn't actually do anything", function () {
+                return request.then(function (response) {
+                    expect(response.body.npi).to.be.an("undefined");
+                });
+            });
+        });
+
+        describe("changing npi as clinician", function () {
+            var request;
+            beforeEach(function () {
+                request = updateUser({role: "clinician" }, { npi: "1245319599" });
+            });
+
+            it("returns a successful response", function () {
+                return expect(request).to.be.a.user.success;
+            });
+            it("does not revoke our access tokens", function () {
+                return request.then(auth.checkTokenWorks(token));
+            });
+            it("stills let us authenticate", function () {
+                return request.then(function () {
+                    return expect(tokenEndpoint(credentials)).to.be.an.api.postSuccess;
+                });
+            });
+
+            it("returns the new npi", function () {
+                return request.then(function (response) {
+                    expect(response.body.npi).to.equal("1245319599");
+                });
+            });
+            it("allows a null npi", function () {
```

Comment:
Do we want the API to accept a null and empty string NPI?

Reply:
The functionality we need is for a clinician to be able to remove their NPI. I'm currently doing this by sending an empty string in the PUT. I allowed null also because first_name last_name and phone allow null, but it's not as necessary

Reply:
That explanation is legit. I'm happy with allowing both of them.